### PR TITLE
Create database singleton

### PR DESCRIPTION
**Pull request by michplunkett**
## Describe Your Changes
Use a [singleton](https://en.wikipedia.org/wiki/Singleton_pattern) design pattern limit distinct connections to the database.

## Checklist Before Requesting a Review
- [x] The code runs successfully.

Verified limited connections (1 for the application and 1 for DataGrip): [SQL statement explanation](https://cloud.ibm.com/docs/databases-for-postgresql?topic=databases-for-postgresql-managing-connections#:~:text=You%20can%20check%20the%20number,down%20the%20connections%20by%20database.)
Screenshot 2023-12-27 at 10 46 57 PM

Tested via session creation and login:
Screenshot 2023-12-27 at 10 41 57 PM

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -11,6 +11,7 @@ cffi==1.16.0 ; os_name == "nt" and implementation_name != "pypy" and python_vers
 charset-normalizer==3.3.2 ; python_version >= "3.11" and python_version < "4.0"
 click==8.1.7 ; python_version >= "3.11" and python_version < "4.0"
 colorama==0.4.6 ; python_version >= "3.11" and python_version < "4.0" and (sys_platform == "win32" or platform_system == "Windows")
+et-xmlfile==1.1.0 ; python_version >= "3.11" and python_version < "4.0"
```
Comment (michplunkett) [inline requirements.txt:14]: Updated the `requirements.txt` file to the current spec.

**Review by FedericoDM — APPROVED** [review]

**Comment by michplunkett** [comment]
I checked the prod database and it is still showing 12 connections. I'm hoping that after the application is cycled sometime in the next day or so, we'll be down to 4 connections since there are only 4 workers.

**Comment by anisfeld** [comment]
This code doesn't work. I'll try to fix it, but this is a good sign we need tests! @FedericoDM 

```
2024-01-03T17:42:24.022525+00:00 app[web.1]:     result = context.run(func, *args)

2024-01-03T17:42:24.022525+00:00 app[web.1]:              ^^^^^^^^^^^^^^^^^^^^^^^^

2024-01-03T17:42:24.022525+00:00 app[web.1]:   File "/app/prijateli_tree/app/routers/games.py", line 153, in view_round

2024-01-03T17:42:24.022526+00:00 app[web.1]:     template_text = languages[player.language.abbr]

2024-01-03T17:42:24.022526+00:00 app[web.1]:                               ^^^^^^^^^^^^^^^

2024-01-03T17:42:24.022526+00:00 app[web.1]:   File "/app/prijateli_tree/app/database.py", line 262, in language

2024-01-03T17:42:24.022526+00:00 app[web.1]:     user = db.query(User).filter_by(id=self.user_id).one()

2024-01-03T17:42:24.022527+00:00 app[web.1]:            ^^^^^^^^

2024-01-03T17:42:24.022527+00:00 app[web.1]: AttributeError: type object 'Database' has no attribute 'query'

2024-01-03T17:42:24.023266+00:00 heroku[router]: at=info method=GET path="/games/6/player/32/round" host=intense-tundra-19419-c4bc16058534.herokuapp.com request_id=896cf634-3184-487a-8062-e9e5945a6ca4 fwd="128.135.204.189" dyno=web.1 connect=0ms service=22ms status=500 bytes=193 protocol=https
```

**Comment by michplunkett** [comment]
I validated it with the adding of sessions, players, etc., so it SHOULD be good. I imagine your problem is a small one line thing the parameter line.

**Pull request merged**